### PR TITLE
Add a setting for disabling rendering "intense" text as bold

### LINKED ISSUE #10576: Add a setting for disabling "intense is bold"
A follow up to #10498

As discussed at length in other threads, esp #109.

Most recently, from @j4james in #10498:

> I mean choose if you want SGR 1 to select a bold font face or not (basically a way to disable this PR). I suppose you could potentially implement that by allowing users to choose which font weight is bold, and expect them to select the normal font weight, but that seems a bit obscure. For example, in XTerm you can just uncheck the "Bold Fonts" menu, and in Konsole you uncheck the option "Draw intense colors in bold font".
> 
> There are a whole bunch of other variations people may want to configure, e.g. disabling bold-as-bright, or only enabling a bold font when used with the extended colors (i.e. the aix bright colors and the ITU 256/RGB colors). Personally I think the latter is the best default config, since it gives you the standard behaviour for the original 8 SGR colors, while still allowing bold fonts when used with extended colors, but I'd be happy with a simple option to turn this off.

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

This adds a new setting `disableIntenseIsBold`. It's a per-appearance, control setting, defaulting to `false`.
* When set to false, we'll render text formatted with `^[[1m` as **bold**. (the 1.10+ default)
* When set to true, we won't. (the pre-1.10 behavior)

## PR Checklist
* [x] Closes #10576 
* [x] I work here
* [ ] Tests added/passed - nah I didn't do that
* [n/a] Requires documentation to be updated

## Validation Steps Performed

image

Yea that works. Printed some bold text, toggled it on, the text was no longer bold. hooray.

### CODE REVIEW

File: src/renderer/dx/DxRenderer.hpp [hunk 1]
```diff
@@ -252,6 +254,7 @@ namespace Microsoft::Console::Render
         // Preferences and overrides
         bool _softwareRendering;
         bool _forceFullRepaintRendering;
+        bool _intenseIsBold{ true };
```
Comment (carlos-zamora) [inline src/renderer/dx/DxRenderer.hpp:257]: nit: looks like all of the other ones are initialized in the ctor. We should probably move this there for consistency

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml [hunk 2]
```diff
@@ -402,6 +402,7 @@
                                           IsChecked="{x:Bind IsBellStyleFlagSet(4), BindBack=SetBellStyleTaskbar, Mode=TwoWay}" />
                             </StackPanel>
                         </local:SettingContainer>
+
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Profiles.xaml:405]: Ah, I see you thought you had to edit Profiles.xaml haha. You can definitely undo this one.

**Review by carlos-zamora — APPROVED** [review]
I'm approving, but please don't forget to...
- [ ] update the schema
- [ ] update the docs

I don't _like_ that the setting is called `disableIntenseIsBold`. I wish we could remove the "disable" part, but then we would have a setting that defaults to `true`, and that's not great either. So imo, it's fine.

**Comment by skyline75489** [comment]
Thanks @zadjii-msft for this! 

Off: this seems to be a good place to also add “intense as bright” option. But I never really know where bright is enabled in DX engine.

**Review by PankajBhojwani — APPROVED** [review]

File: src/renderer/dx/DxRenderer.hpp (hunk 1 above)
Comment (DHowett) [reply src/renderer/dx/DxRenderer.hpp]: We should start moving these things into the header honestly.

**Comment by zadjii-msft** [comment]
I'm gonna close this PR, as to not pollute this PR that's already got green checks. From discussion in sync today - we're not gonna implement the setting this way right now. We're instead gonna do the EnumFlagMapper thing, and do `"intenseAs": "none"|"bold"|"bright"|"all"` (naming tbd). I'm not gonna touch the "bright" setting for now, and just slam in the "bold"/"none" one.

**Pull request closed without merging**